ktor-client/ktor-client-core/common/src/io/ktor/client/HttpClient.kt

```diff
@@ -154,17 +154,17 @@ public class HttpClient(
                 config.install("DefaultTransformers") { defaultTransformers() }
             }
 
-            if (expectSuccess) {
-                config.addDefaultResponseValidation()
-            }
-
             config.install(HttpSend)
 
             if (followRedirects) {
                 config.install(HttpRedirect)
             }
 
             config += this
+
+            // it's important to install validators last
```

Comment:
Could we fix that by inserting a new phase? Please remove the comment

Reply:
No, we can't, because validation happens not in pipeline interceptor, but `HttpSend` interceptor.
Comment removed.